src/cli_parser.cpp

```diff
@@ -330,35 +371,39 @@ void CLIParser::prepare(ServerSettingsImpl* serverSettings, ModelsSettingsImpl*
-    if (result->count("log_path"))
-        serverSettings->logPath = result->operator[]("log_path").as<std::string>();
-    #if (PYTHON_DISABLE == 0)
-        serverSettings->withPython = true;
-    #endif
-
-#ifdef MTR_ENABLED
-    if (result->count("trace_path"))
-        serverSettings->tracePath = result->operator[]("trace_path").as<std::string>();
-#endif
-
-    if (result->count("grpc_channel_arguments"))
-        serverSettings->grpcChannelArguments = result->operator[]("grpc_channel_arguments").as<std::string>();
-
-    serverSettings->filesystemPollWaitMilliseconds = result->operator[]("file_system_poll_wait_seconds").as<uint32_t>() * 1000;
-    serverSettings->sequenceCleanerPollWaitMinutes = result->operator[]("sequence_cleaner_poll_wait_minutes").as<uint32_t>();
-    serverSettings->resourcesCleanerPollWaitSeconds = result->operator[]("custom_node_resources_cleaner_interval_seconds").as<uint32_t>();
-
-    if (result != nullptr && result->count("cache_dir")) {
-        serverSettings->cacheDir = result->operator[]("cache_dir").as<std::string>();
-    }
-
     if (result->count("config_path")) {
         modelsSettings->configPath = result->operator[]("config_path").as<std::string>();
         modelsSettings->userSetSingleModelArguments.push_back("config_path");
     }
-    serverSettings->startedWithCLI = true;
+
+    // Ovms Pull models mode
+    if (result->count("pull")) {
+        serverSettings->hfSettings.pullHfModelMode = result->operator[]("pull").as<bool>();
+        if (result->count("overwrite_models"))
+            serverSettings->hfSettings.overwriteModels = result->operator[]("overwrite_models").as<bool>();
+        if (result->count("source_model"))
+            serverSettings->hfSettings.sourceModel = result->operator[]("source_model").as<std::string>();
+        if (result->count("model_repository_path"))
+            serverSettings->hfSettings.downloadPath = result->operator[]("model_repository_path").as<std::string>();
+        if (result->count("task")) {
+            std::string task = result->operator[]("task").as<std::string>();
+            serverSettings->hfSettings.task = task;
+            if (task == "text_generation") {
+                this->graphOptionsParser.prepare(serverSettings, modelsSettings);
+            } else if (task == "embeddings") {
+                this->embeddingsGraphOptionsParser.prepare(serverSettings, modelsSettings);
+            } else if (task == "rerank") {
+                this->rerankGraphOptionsParser.prepare(serverSettings, modelsSettings);
+            } else {
+                throw std::logic_error("Error: --task parameter unsupported value: " + task);
+            }
+        } else {
+            // Default text_generation task
+            this->graphOptionsParser.prepare(serverSettings, modelsSettings);
+        }
+    } else {
+        serverSettings->hfSettings.pullHfModelMode = false;
+    }
```

Comment:
I think we can split this into 3?
1) prepare sharedServerSettings(serverSettings&)
2) prepare modelSettings(modelsSettings&)
3) prepare pullModeSettings(hfSettings&)